File: app/src/main/java/CS399/OnRead/writeJson.java
package CS399.OnRead;

import android.content.Context;
import android.media.Image;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import java.io.FileOutputStream;

public class writeJson {


    //values from input page, id values are random and need to be changed later based on actual id values in input file

    EditText contactName;

    EditText contactNumber;

    EditText timeThreshold;

    EditText message;

    Button writeJsonFile;


    public void writeJson(View view) {

        contactName = ( EditText ) findViewById(R.id.contactName);
        contactNumber = (EditText) findViewById( R.id.contactNumber)
        timeThreshold = (EditText) findViewById( R.id.timeThreshold);
        message = (EditText) findViewById( R.id.message );
        writeJsonFile = (Button) findViewById(R.id.writeJsonFile);


        writeJsonFile.setOnClickListener(new View.OnClickListener())
        {
            @Override
            public void onClick(View view)
            {

                save(main.Assets.bots, contactName, contactNumber, timeTreshold, message);
            }
        }

    }

    public void save(String inputFile, String name , String phone, String time, String message)
    {
        try
        {
            FileOutputStream file = openFileOutput("bots.json", Context.MODE_PRIVATE);
            file.write(name + phone +  time + message);
            file.close();

        }
        catch( Exception exc)
        {
            Toast.makeText(writeJson.this, "Error.", Toast.LENGTH_LONG).show();
        }
    }
}

File: app/src/main/java/CS399/OnRead/writeJson.kt
package CS399.OnRead

import android.content.Context
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class writeJson(view: View?) : AppCompatActivity() {
    //values from input page, id values are random and need to be changed later based on actual id values in input file
    var contactName: EditText
    var contactNumber: EditText
    var timeThreshold: EditText
    var message: EditText
    var writeJsonFile: Button
    fun save(
        inputFile: String?,
        name: String?,
        phone: String?,
        time: String?,
        message: String?
    ) {
        try {
            val file =
                openFileOutput("bots.json", Context.MODE_PRIVATE)
            file.write(contactName + contactNumber + timeTreshold + textMessage)
            file.close()
        } catch (exc: Exception) {
            Toast.makeText(this@writeJson, "Error.", Toast.LENGTH_LONG).show()
        }
    }

    init {
        contactName = findViewById<View>(R.id.contactName) as EditText
        contactNumber = findViewById<View>(R.id.contactNumber) as EditText
        timeThreshold = findViewById<View>(R.id.timeThreshold) as EditText
        message = findViewById<View>(R.id.textMessage) as EditText
        writeJsonFile = findViewById<View>(R.id.writeJsonFile) as Button
        writeJsonFile.setOnClickListener {
            save(
                "bots.json",
                contactName.text.toString(),
                contactNumber.text.toString(),
                timeTreshold,
                textMessage
            )
        }
    }
}